BusinessLogic/Entities/Auth/OAuthClientPassword.cs

```diff
@@ -117,15 +117,30 @@ public async Task<bool> SendEvent(Event e, Subscription subscription)
         /// <param name="response"></param>
         private void ResetAccessTokenIfErrorInResponseMessage(HttpResponseMessage response)
         {
-            List<Dictionary<string, string>> resultValues = JsonConvert.DeserializeObject<List<Dictionary<string, string>>>(response.Content.ReadAsStringAsync().Result);
 
-            if (resultValues.Count > 0)
+            // if error code string is there, then parse
+            if (response.Content.ReadAsStringAsync().Result.IndexOf("errorCode") != -1)
```

Comment:
```suggestion
            if ((await response.Content.ReadAsStringAsync()).IndexOf("errorCode") != -1)
```
Instead of using `.Result` here you can use `await` as shown above

Reply:
The await is outside on line 92, the function only receives the result